dtfs-central-api/src/helpers/get-keying-sheet-calculation-tfm-facility-values.test.ts

```diff
@@ -0,0 +1,173 @@
+        new NotFoundError(`Failed to find a cover end date for the tfm facility with ukef facility id '${facilityId}`),
+      );
+    });
+  });
+
+  describe('when the report period is provided', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue(mockFacility);
+    });
+
+    describe('when the facility does not contain a completed amendment', () => {
+      it("should return a populated object with the facility's cover end date", async () => {
+        const result = await getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod);
+
+        const expected = {
+          coverEndDate: convertTimestampToDate(today),
+          coverStartDate: convertTimestampToDate(today),
+          dayCountBasis: 365,
+          interestPercentage: 5,
+          coverPercentage: 5,
+          value: mockFacility.facilitySnapshot.value,
+        };
+
+        expect(result).toEqual(expected);
+      });
+    });
+
+    describe('when the facility contains a completed amendment', () => {
+      const dateAfterReportPeriodEnd = new Date(addDays(today, 1));
+
+      const mockFacilityWithAmendment = {
+        ...mockFacility,
+        amendments: [
+          {
+            ...aCompletedTfmFacilityAmendment(),
+            value: 350000,
+            effectiveDate: dateAfterReportPeriodEnd.getTime(),
+            // 365 days after report period end
+            coverEndDate: new Date(addDays(today, 365)).getTime(),
+          },
+        ],
+      };
+      beforeEach(() => {
+        findOneByUkefFacilityIdSpy.mockResolvedValue(mockFacilityWithAmendment);
+      });
+
+      it("should return a populated object with the amendment's cover end date", async () => {
+        const result = await getKeyingSheetCalculationTfmFacilityValues(facilityId, reportPeriod);
+
+        const expected = {
+          coverEndDate: convertTimestampToDate(mockFacilityWithAmendment.amendments[0].coverEndDate),
+          coverStartDate: convertTimestampToDate(today),
+          dayCountBasis: 365,
+          interestPercentage: 5,
+          coverPercentage: 5,
+          value: mockFacility.facilitySnapshot.value,
+        };
+
+        expect(result).toEqual(expected);
+      });
+    });
+  });
+
+  describe('when the report period is not provided', () => {
+    beforeEach(() => {
+      findOneByUkefFacilityIdSpy.mockResolvedValue(mockFacility);
+    });
+
+    it("should return a populated object with the facility's cover end date", async () => {
+      const result = await getKeyingSheetCalculationTfmFacilityValues(facilityId);
+
+      const expected = {
+        coverEndDate: convertTimestampToDate(today),
+        coverStartDate: convertTimestampToDate(today),
+        dayCountBasis: 365,
+        interestPercentage: 5,
+        coverPercentage: 5,
+        value: mockFacility.facilitySnapshot.value,
+      };
+
+      expect(result).toEqual(expected);
+    });
+  });
```

Comment:
Need to add the with amendments case here please

Reply:
done